src/Integration/MefServices/DeprecatedSonarRuleSetManager.cs

```diff
@@ -0,0 +1,171 @@
+            "*****************************************************************************************";
+
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+        private readonly IActiveSolutionTracker activeSolutionTracker;
+        private readonly IProjectSystemHelper projectSystemHelper;
+        private readonly ISolutionRuleSetsInformationProvider ruleSetProvider;
+        private readonly ILogger logger;
+
+        private bool isDisposed;
+
+        [ImportingConstructor]
+        public DeprecatedSonarRuleSetManager(IHost host)
+            : this(host.GetMefService<IActiveSolutionBoundTracker>(), host.GetMefService<IActiveSolutionTracker>(),
+                host.GetService<IProjectSystemHelper>(), host.GetService<ISolutionRuleSetsInformationProvider>(), host.Logger)
+        {
+        }
+
+        internal /* for testing purposes */ DeprecatedSonarRuleSetManager(IActiveSolutionBoundTracker activeSolutionBoundTracker,
+            IActiveSolutionTracker activeSolutionTracker, IProjectSystemHelper projectSystemHelper,
+            ISolutionRuleSetsInformationProvider ruleSetProvider, ILogger logger)
+        {
+            if (activeSolutionBoundTracker == null)
+            {
+                throw new ArgumentNullException(nameof(activeSolutionBoundTracker));
+            }
+
+            if (activeSolutionTracker == null)
+            {
+                throw new ArgumentNullException(nameof(activeSolutionTracker));
+            }
+
+            if (projectSystemHelper == null)
+            {
+                throw new ArgumentNullException(nameof(projectSystemHelper));
+            }
+
+            if (ruleSetProvider == null)
+            {
+                throw new ArgumentNullException(nameof(ruleSetProvider));
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            this.activeSolutionTracker = activeSolutionTracker;
+            this.projectSystemHelper = projectSystemHelper;
+            this.ruleSetProvider = ruleSetProvider;
+            this.logger = logger;
+
+            this.activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
+            this.activeSolutionTracker.ActiveSolutionChanged += OnActiveSolutionChanged;
+
+            if (this.activeSolutionBoundTracker.CurrentConfiguration != null &&
+                this.activeSolutionBoundTracker.CurrentConfiguration.Mode != NewConnectedMode.SonarLintMode.LegacyConnected)
+            {
+                WarnIfAnyProjectHasSonarRuleSet();
+            }
+        }
+
+        private void OnActiveSolutionChanged(object sender, ActiveSolutionChangedEventArgs e)
+        {
+            if (e.IsSolutionOpen)
+            {
+                WarnIfAnyProjectHasSonarRuleSet();
+            }
```

Comment:
The event handlers will be called on the UI thread so it would be safer to have a _try-catch-squash-if-not-critical block_.